scripts/validate_rst_title_capitalization.py

```diff
@@ -184,12 +189,20 @@ def correct_title_capitalization(title: str) -> str:
     # first word character.
     correct_title: str = re.sub(r"^\W*", "", title).capitalize()
 
-    # Remove a URL from the title. We do this because words in a URL must
-    # stay lowercase, even if they are a capitalization exception.
-    removed_https_title = re.sub(r"<https?:\/\/.*[\r\n]*>", "", correct_title)
-
     # Split a title into a list using non-word character delimiters.
-    word_list = re.split(r"\W", removed_https_title)
+    word_list = re.split(r"\W", correct_title)
+
+    # Recombine hyphenated words
+    for word in correct_title.split():
+        if '-' in word:
+            lst = word.split('-')
+            first = lst[0]
+            for idx, val in enumerate(word_list):
+                if val == first:
+                    for _ in range(len(lst)):
+                        del word_list[idx]
+                    word_list.insert(idx, '-'.join(lst))
+                    break
```

Comment:
```suggestion
    word_list = re.split(r"[^a-zA-Z0-9_-]", correct_title)
```

I think by changing the regular expression you can get rid of this function. I think the regex `[\W-]` should also work and it's way shorter, but in a quick test it didn't seem like it worked, but feel free to research a better pattern.